### PR TITLE
feature(ManagedSpy): Make defined message go through processes at vis…

### PR DESCRIPTION
**Pull request by uian2000**
…ta and upper

Import ChangeWindowMessageFilter from user32.dll
Register user defined message(copy and modified from Messages.h of ManagedSpyLib)
Make these messages pass through with ChangeWindowMessageFilter

Signed-off-by: uian2000 <uian2000@gmail.com>

### CODE REVIEW

**Comment by ForNeVeR** [comment]
That's very interesting PR. Could you please elaborate a bit? What's the purpose of this change? What particular task does it solve?

File: ManagedSpy/MainForm.cs [hunk 1]
```diff
@@ -20,12 +21,67 @@ public partial class MainForm : Form
 		/// </summary>
 		private ControlProxy currentProxy = null;
 		EventFilterDialog dialog = new EventFilterDialog();
+        /// <summary>
+        /// Modified by uian2000@gmail.com
```
Comment (ForNeVeR) [inline ManagedSpy/MainForm.cs]: Please no need to sign-off every source line you modify. You will anyways be highlighted as a project contributor on [this page](https://github.com/ForNeVeR/ManagedSpy/graphs/contributors). Please remove your email from these comments. Thanks!

File: ManagedSpy/MainForm.cs [hunk 2]
```diff
@@ -20,12 +21,67 @@ public partial class MainForm : Form
...
+        /// Import ChangeWindowMessageFilter/RegisterWindowMessage to allow message exchange between processes with different priviliage.
+        /// </summary>
+        [DllImport("user32")]
+        public static extern bool ChangeWindowMessageFilter(uint msg, int flag);
+        [DllImport("user32")]
+        public static extern uint RegisterWindowMessage(string lpString);
```
Comment (ForNeVeR) [inline ManagedSpy/MainForm.cs]: I also don't think that's a good architectural solution to keep these methods in the `MainForm`. I _suggest_ to remove these to a new class. But if you have any problems with that, I can do it myself.

Comment (uian2000) [inline ManagedSpy/MainForm.cs]: Yes please make it.
I've tested managedspy on my win7 and win10, no message appeared before I've make this ChangeWindowMessageFilter.
I'm newbee of dot net framework, and this is just a little trick make it go.
Please make graceful refactor, that I can learn from it.

File: ManagedSpy/MainForm.cs (hunk 1 above)
Comment (uian2000) [inline ManagedSpy/MainForm.cs]: I don't know how to remove it.
I make use of git locally, this is first time I've rise pull request.

Comment (ForNeVeR) [inline ManagedSpy/MainForm.cs]: That's simple: just remove these lines from the source, commit and push them to your repository.

If you really have any problem with that, I'm ready to delete these myself. Are you okay with that?

**Comment by ForNeVeR** [comment]
I think I got it. The comment in `MainForm.cs` states that this PR should make ManagedSpy to work with differently-privileged processes. I'm going to check that and probably merge the PR afterwards.

**Comment by uian2000** [comment]
For some detail, This PR is managed to resolve UIPI probloms.
See this description: https://msdn.microsoft.com/zh-cn/library/ee663086.aspx

File: ManagedSpy/MainForm.cs (hunk 2 above)
Comment (ForNeVeR) [inline ManagedSpy/MainForm.cs]: After some thoughts, I've decided to move the code in question to the static `MessageFilters` class on C++/CLI side (i.e. in ManagedSpyLib). See commit https://github.com/ForNeVeR/ManagedSpy/pull/18/commits/f48fd85ca91a60e7955af9575543c736202ab59a.

**Comment by ForNeVeR** [comment]
@uian2000 thank you much for your work. I've added the issue #17 to track UIPI behavior and created PR #18 with refactoring based on your code. Your changes are now incorporated into the main branch; I'll publish the new ManagedSpy release shortly.

**Pull request merged**